### PR TITLE
[#4785] Refactor Kafka Plugin and Add ConsumerRecords Interceptor.

### PR DESCRIPTION
**Pull request by VictorZeng**
Title: Kafka Plugin Contribution (optimize)

Link: https://github.com/VictorZeng/pinpoint/tree/kafka-plugin/plugins/kafka
Target: pinpoint-kafka-plugin
Supported Version: kafka version 0.11+.
Description: Refactor Kafka Plugin and add ConsumerRecords Interceptor. If you consume from the same topic and want to intercept batch consumption methods.

Add Annotations:

-  (143, "kafka.batch", VIEW_IN_RECORD_SET)

### CODE REVIEW

**Comment by VictorZeng** [comment]
fix issue #4785

**Comment by VictorZeng** [comment]
PTAL @Xylus @koo-taejin

**Comment by VictorZeng** [comment]
image

**Comment by koo-taejin** [comment]
hi @VictorZeng 

Thank you for your pull request.
Let me take a look.

thanks. :)

**Comment by VictorZeng** [comment]
Hi @koo-taejin Can merge this PR？

File: plugins/kafka/src/main/java/com/navercorp/pinpoint/plugin/kafka/interceptor/ConsumerMultiRecordEntryPointInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    public ConsumerMultiRecordEntryPointInterceptor(TraceContext traceContext, MethodDescriptor methodDescriptor) {
+        super(traceContext, methodDescriptor);
+    }
+
+    @Override
+    protected Trace createTrace(Object target, Object[] args) {
+
+        ConsumerRecords consumerRecords = getConsumerRecords(args);
+
+        if (consumerRecords == null || consumerRecords.isEmpty())
```
Comment (koo-taejin) [inline plugins/kafka/src/main/java/com/navercorp/pinpoint/plugin/kafka/interceptor/ConsumerMultiRecordEntryPointInterceptor.java]: please use curly bracket.

File: plugins/kafka/src/main/java/com/navercorp/pinpoint/plugin/kafka/interceptor/ConsumerRecordEntryPointInterceptor.java [hunk 2]
```diff
@@ -41,53 +34,46 @@
...
         ConsumerRecord consumerRecord = getConsumerRecord(args);
-        if (consumerRecord == null) {
-            return;
-        }
 
-        Trace trace = getTrace(consumerRecord);
-        if (trace == null) {
-            return;
-        }
+        if (consumerRecord == null)
```
Comment (koo-taejin) [inline plugins/kafka/src/main/java/com/navercorp/pinpoint/plugin/kafka/interceptor/ConsumerRecordEntryPointInterceptor.java]: please use curly bracket.

File: plugins/kafka/src/main/java/com/navercorp/pinpoint/plugin/kafka/interceptor/ConsumerMultiRecordEntryPointInterceptor.java (hunk 1 above)
Comment (VictorZeng) [reply plugins/kafka/src/main/java/com/navercorp/pinpoint/plugin/kafka/interceptor/ConsumerMultiRecordEntryPointInterceptor.java]: OK

File: plugins/kafka/src/main/java/com/navercorp/pinpoint/plugin/kafka/interceptor/ConsumerRecordEntryPointInterceptor.java (hunk 2 above)
Comment (VictorZeng) [reply plugins/kafka/src/main/java/com/navercorp/pinpoint/plugin/kafka/interceptor/ConsumerRecordEntryPointInterceptor.java]: OK

**Comment by koo-taejin** [comment]
@VictorZeng 

Please add commnet for user. 
If user include ConsumerRecords in profiler.kafka.consumer.entryPoint in pinpoint.config.
then Remote Trace feature is not enabled.

I will merge this pr if you work on the above curly bracket and this part.

Thanks again for you're pull request .

**Comment by VictorZeng** [comment]
@koo-taejin Thank U so much.

**Comment by koo-taejin** [comment]
@VictorZeng 

Please multiple commits into one  using rebase. 
thanks.

**Comment by VictorZeng** [comment]
@koo-taejin OK, I will pull a new request.

**Comment by VictorZeng** [comment]
@koo-taejin why has conflicts...?

**Comment by koo-taejin** [comment]
@VictorZeng 

It looks like something is wrong.

If you look at Files changed, it seems that your work and your not work have changed.

**Comment by VictorZeng** [comment]
@koo-taejin I think create a new branch and pull a  new request ...

**Comment by VictorZeng** [comment]
continuation from #4808

**Pull request closed without merging**